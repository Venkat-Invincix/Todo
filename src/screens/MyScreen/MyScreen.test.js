import 'react-native';
import React from 'react';
import { create, act } from 'react-test-renderer';
import Myscreen from './MyScreen';
import { Text, View } from 'react-native';

const tree = create(<Myscreen />)

describe('MyScreen Testing', () => {

    it('snapshot', () => {
        expect(tree).toMatchSnapshot();
    });

    it('Button Press', () => {

        const button = tree.root.findByProps({ testID: 'myButton' }).props;
        // console.log(button);
        act(() => button.onPress())
        const text = tree.root.findByProps({ testID: 'myText' }).props;
        expect(text.children).toBe('button pressed');

    });

    it('To Know', () => {
        const text = 'bar';
        const instance = create(
            <View>
                <Text testID='foo'>{text}</Text>
            </View>
        ).root;

        const result = instance.findByProps({ testID: 'foo' }).children.length

        // console.log('result', result);
        expect(result).toBe(1);
    });


});
