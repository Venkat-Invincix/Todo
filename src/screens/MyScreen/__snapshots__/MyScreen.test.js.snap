// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyScreen Testing snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "green",
    }
  }
>
  <Text
    testID="myText"
  >
    
  </Text>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="myButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Press Me
      </Text>
    </View>
  </View>
</View>
`;
